pkg/probe/grpc/grpc.go

```diff
@@ -47,14 +47,16 @@ func New() Prober {
 
 // Probe executes a grpc call to check the liveness/readiness/startup of container.
 // Returns the Result status, command output, and errors if any.
-// Only return non-nil error when service is unavailable and/or not implementing the interface,
-// otherwise result status is failed,BUT err is nil
-func (p grpcProber) Probe(host, service string, port int, timeout time.Duration, opts ...grpc.DialOption) (probe.Result, string, error) {
+// Any failure is considered as a probe failure to mimic grpc_health_probe tool behavior.
+// err is always nil
+func (p grpcProber) Probe(host, service string, port int, timeout time.Duration) (probe.Result, string, error) {
 	v := version.Get()
 
-	md := metadata.New(map[string]string{
-		"User-Agent": fmt.Sprintf("kube-probe/%s.%s", v.Major, v.Minor),
-	})
+	opts := []grpc.DialOption{
+		grpc.WithUserAgent(fmt.Sprintf("kube-probe/%s.%s", v.Major, v.Minor)),
+		grpc.WithBlock(),
+		grpc.WithInsecure(), //credentials are currently not supported
+	}
```

Comment:
does this mean that the probe will always have this hardcoded options?
do we have some grpc experienced people that can advice if this cover most of the use cases?

Reply:
Followed https://github.com/grpc-ecosystem/grpc-health-probe/blob/b0cf9b4985ab201654da993ea90a7b459527fadb/main.go#L216-L219

it doesn't make sense to set up options from the prober - all grpc specifics must stay in this file. So I removed an `opts` arguments

Reply:
no idea , just why I asked :smile:

Reply:
I was just explaining that this clean up and those options DO make sense. Basically I am doing all the same as grpc_health_probe that is commonly used would do. So people can just swap from grpc_health_probe to native grpc support without any behavior change.

Reply:
Sounds good, it also seems aligned with the Https probe behaviour

>  If scheme field is set to HTTPS, the kubelet sends an HTTPS request skipping the certificate verificatio

Thanks for the insight